### DISCUSSION TITLE
Support Cloud Volumes per node pool

### DISCUSSION THREAD

**Discussion by Funzinator** [vitobotta/hetzner-k3s#291] [Ideas]
Is there an easy way, for example, to provision all masters with a Hetzner Cloud volume automatically?
I am thinking about using [Longhorn](https://longhorn.io/) for smaller volume needs.
Hetzner requires a minimum of 10 Gi per volume but sometimes you just want so persisted small space (e.g., for loki, fluent-bit, redis) where 10 Gi are just too much.

I think it would be nice to specify volumes per node pool (and/or for the master nodes) so you can have dedicated storage nodes.

WDYT?

**Comment by vitobotta** [comment]
I haven't thought about attaching volumes, I'd need to check the API and see how to do it. But yeah the min 10GB is quite annoying. Longhorn is great in that it does thin provisioning, so you can save money with it. On a couple of clusters I use Longhorn and it works nicely.

**Reply by Funzinator** [reply]
Right, thin provisioning and over-commitment are quite handy. The Hetzner volumes are already triple-replicated according to the docs, so one could use longhorn even with 2 replicas and still be on the safe side (replication-wise) and save money.

I'll also have a look, but I don't know anything about Crystal...
